checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -0,0 +1,1602 @@
+              formatMissingMustCallMethods(mustCallValue),
+              firstlocalVarWithTree.localVar.getType().toString(),
+              outOfScopeReason);
+        }
+      }
+    }
+  }
+
+  /**
+   * Increment the -AcountMustCall counter.
+   *
+   * @param node the node being counted, to extract the type
+   */
+  private void incrementNumMustCall(Node node) {
+    if (checker.hasOption(ResourceLeakChecker.COUNT_MUST_CALL)) {
+      TypeMirror type = node.getType();
+      incrementMustCallImpl(type);
+    }
+  }
+
+  /**
+   * Increment the -AcountMustCall counter.
+   *
+   * @param elt the elt being counted, to extract the type
+   */
+  private void incrementNumMustCall(Element elt) {
+    if (checker.hasOption(ResourceLeakChecker.COUNT_MUST_CALL)) {
+      TypeMirror type = elt.asType();
+      incrementMustCallImpl(type);
+    }
+  }
+
+  /**
+   * Shared implementation for the two version of countMustCall. Don't call this directly.
+   *
+   * @param type the type of the object that has a must call obligation
+   */
+  private void incrementMustCallImpl(TypeMirror type) {
+    // only count uses of JDK classes, since that's what we report on in the paper
+    String qualifiedName = TypesUtils.getTypeElement(type).getQualifiedName().toString();
+    if (!qualifiedName.startsWith("java")) {
+      return;
+    }
+    checker.numMustCall++;
+  }
+
+  /**
+   * Do the called methods represented by the {@link CalledMethods} type {@code cmAnno} include all
+   * the methods in {@code mustCallValues}?
+   *
+   * @param mustCallValues the strings representing the must-call obligations
+   * @param cmAnno an annotation from the called-methods type hierarchy
+   * @return true iff cmAnno is a subtype of a called-methods annotation with the same values as
+   *     mustCallValues
+   */
+  private boolean calledMethodsSatisfyMustCall(
+      List<String> mustCallValues, AnnotationMirror cmAnno) {
+    AnnotationMirror cmAnnoForMustCallMethods =
+        typeFactory.createCalledMethods(mustCallValues.toArray(new String[0]));
+    return typeFactory.getQualifierHierarchy().isSubtype(cmAnno, cmAnnoForMustCallMethods);
+  }
+
+  /**
+   * Is {@code exceptionClassName} an exception type we are ignoring, to avoid excessive false
+   * positives? For now we ignore {@code java.lang.Throwable}, {@code NullPointerException}, and the
+   * runtime exceptions that can occur at any point during the program due to something going wrong
+   * in the JVM, like OutOfMemoryErrors or ClassCircularityErrors.
+   *
+   * @param exceptionClassName the fully-qualified name of the exception
+   * @return true if the given exception class should be ignored
+   */
+  private static boolean isIgnoredExceptionType(@FullyQualifiedName Name exceptionClassName) {
+    // any method call has a CFG edge for Throwable/RuntimeException/Error to represent run-time
+    // misbehavior. Ignore it.
+    return exceptionClassName.contentEquals(Throwable.class.getCanonicalName())
```

Comment:
`contentEquals` is expensive (this fact was a surprise to me when I learned it).
I suggest converting to a string, then checking for containment in a HashSet.  These comments will still be useful when constructing the HashSet.